src/module_wrap.cc

```diff
@@ -255,215 +257,219 @@ MaybeLocal<Module> ModuleWrap::ResolveCallback(Local<Context> context,
     return MaybeLocal<Module>();
   }
 
-  ModuleWrap* mod;
-  ASSIGN_OR_RETURN_UNWRAP(&mod, module_object, MaybeLocal<Module>());
-  return mod->module_.Get(env->isolate());
+  ModuleWrap* module;
+  ASSIGN_OR_RETURN_UNWRAP(&module, module_object, MaybeLocal<Module>());
+  return module->module_.Get(env->isolate());
 }
 
 namespace {
 
-URL __init_cwd() {
-  std::string specifier = "file://";
-#ifdef _WIN32
-  // MAX_PATH is in characters, not bytes. Make sure we have enough headroom.
-  char buf[MAX_PATH * 4];
-#else
-  char buf[PATH_MAX];
-#endif
-
-  size_t cwd_len = sizeof(buf);
-  int err = uv_cwd(buf, &cwd_len);
-  if (err) {
-    return URL("");
-  }
-  specifier += buf;
-  specifier += "/";
-  return URL(specifier);
-}
-static URL INITIAL_CWD(__init_cwd());
-inline bool is_relative_or_absolute_path(std::string specifier) {
+// Tests whether a path starts with /, ./ or ../
+inline bool PathIsBasedOffCurrentDirectory(const std::string& specifier) {
```

Comment:
Note that this is a URL-related function, designed to mimic the resolution steps under https://html.spec.whatwg.org/#integration-with-the-javascript-module-system.

I think _isRelativeOrAbsolute_ remains a good name here.

Reply:
Also it seems like you're flipping the polarity of this function? If that is the case perhaps a better name might be _isBareSpecifier_ to match the WhatWG terminology.

Reply:
Although since this function relies on `parseAsURL` already being checked, _isNotRelativeOrAbsolute_ may be the most descriptive!

Reply:
@guybedford I can’t find what exactly you’re referring to in that spec (also, linking to that page is a great way to DoS somebody’s browser)

At least outside of web specs, relative paths are the paths that are not absolute, so `IsRelativeOrAbsolute` really is not a helpful name?

Maybe `IsBareSpecifier` works, but even that isn’t really saying anything about what’s meant by it … I can add a comment indicating what this matches in WhatWG terminology, but the part of the goal of this change is to make it more accessible to people who don’t know all the spec phrases.

Reply:
@addaleax here's the appropriate spec text, which is what is exactly what the function here is distinguishing:

```
To resolve a module specifier given a script script and a JavaScript string specifier, perform the following steps. It will return either a URL record or failure.

1. Apply the URL parser to specifier. If the result is not failure, return the result.

2. If specifier does not start with the character U+002F SOLIDUS (/), the two-character sequence U+002E FULL STOP, U+002F SOLIDUS (./), or the three-character sequence U+002E FULL STOP, U+002E FULL STOP, U+002F SOLIDUS (../), return failure and abort these steps.

3. This restriction is in place so that in the future we can allow custom module loaders to give special meaning to "bare" import specifiers, like import "jquery" or import "web/crypto". For now any such imports will fail, instead of being treated as relative URLs.

3. Return the result of applying the URL parser to specifier with script's base URL as the base URL.
```

`IsNotRelativeOrAbsolute` is trying to capture the concept of what is remaining being a plain or bare name (eg `lodash`).

Note that for example, we do not check `.\` or handle internal `..` segments, we are exactly just checking the above features. That is "bare names" are detected identically to the WhatWG URL spec, which then get resolved as packages.

It's nits though certainly.

Reply:
Yeah, but this bikeshedding might be worth it. :)

So how does something like `ShouldBeTreatedAsRelativeOrAbsolutePath` sound to you? Does that capture the essence of the semantics well?

Reply:
That sounds about right, yes. Although with the function polarity change, I think it would make it something like `ShouldNotBeTreatedAsRelativeOrAbsolutePath`? My comment below then may make more sense as well in terms of it being that confusing concept to humans of the double negative :)

Reply:
I mean, we can just change polarity back … ;)